Orange/widgets/widget.py

```diff
@@ -539,6 +539,48 @@ def progressBarFinished(self, processEvents=QEventLoop.AllEvents):
         if processEvents is not None and processEvents is not False:
             qApp.processEvents(processEvents)
 
+    def progressBar(self, show_bar=True):
```

Comment:
The current _preferred_ progress bar interface is to use `gui.progressBar` which provides a nice `advance()` wrapper for when the number of steps is known in advance.

Could this method accept instead a number of steps (like the one in `gui` does) and expose a similar method on the in-`__enter__`-returned `ProgressBarHandler` object?